src/qt/walletcontroller.cpp

```diff
@@ -46,13 +46,15 @@ std::vector<WalletModel*> WalletController::getWallets() const
     return m_wallets;
 }
 
-std::vector<std::string> WalletController::getWalletsAvailableToOpen() const
+std::map<std::string, bool> WalletController::listWalletDir() const
 {
     QMutexLocker locker(&m_mutex);
-    std::vector<std::string> wallets = m_node.listWalletDir();
+    std::map<std::string, bool> wallets;
+    for (const std::string& name : m_node.listWalletDir()) {
+        wallets[name] = false;
+    }
     for (WalletModel* wallet_model : m_wallets) {
-        auto it = std::remove(wallets.begin(), wallets.end(), wallet_model->wallet().getWalletName());
-        if (it != wallets.end()) wallets.erase(it);
+        wallets[wallet_model->wallet().getWalletName()] = true;
```

Comment:
I think this needs to check if  `wallet_model->wallet().getWalletName()` is in the map in the first place before setting it to true, in case there's a wallet loaded (for whatever reason) that's not in the wallet dir

Reply:
Good catch.

Reply:
> for whatever reason

One reason is external wallets.

Reply:
Fixed.